### PR TITLE
Refactor ios FacelLandmarker

### PR DESCRIPTION
**Pull request by st-tuanmai**
### Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

Fixes # (issue)

### Checklist
Please ensure the following items are complete before submitting a pull request:

- [x] My code follows the code style of the project.
- [ ] I have updated the documentation (if applicable).
- [x] I have added tests to cover my changes.

### Type of Change
Please check the relevant option below:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update (non-breaking change which updates documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Screenshots
If applicable, please add screenshots to help explain your changes.

### Additional Notes
Add any additional information or context about the pull request here.

### CODE REVIEW

**Comment by khanhlvg** [comment]
@priankakariatyml Can you take a look? Thanks!

File: examples/face_landmarker/ios/FaceLandmarker/Configs/InferenceConfigurationManager.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 The MediaPipe Authors.
```
Comment (priankakariatyml) [inline examples/face_landmarker/ios/FaceLandmarker/Configs/InferenceConfigurationManager.swift:1]: Rename folder to 'Configurations'

File: examples/face_landmarker/ios/FaceLandmarker/Views/FaceOverlay.swift [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+  let color: UIColor
+  let lines: [Line]
+}
+
+/**
+ This structure holds the display parameters for the overlay to be drawon on a detected object.
+ */
+struct FaceOverlay {
+  let dots: [CGPoint]
+  let lineConnectios: [LineConnection]
```
Comment (priankakariatyml) [inline examples/face_landmarker/ios/FaceLandmarker/Views/FaceOverlay.swift:35]: s/lineConnectios/lineConnections

File: examples/face_landmarker/ios/FaceLandmarker/Views/FaceOverlay.swift [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+}
+
+/// Custom view to visualize the face landmarks result on top of the input image.
```
Comment (priankakariatyml) [inline examples/face_landmarker/ios/FaceLandmarker/Views/FaceOverlay.swift:38]: s/landmarks result/landmarker result

File: examples/face_landmarker/ios/FaceLandmarker/Views/FaceOverlay.swift [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+      height: imageSize.height * scaleFactor)
+    let xOffset = (viewSize.width - scaledSize.width) / 2
+    let yOffset = (viewSize.height - scaledSize.height) / 2
+
+    return (xOffset, yOffset, scaleFactor)
+  }
+
+  // Helper to get object overlays from detections.
+  static func faceOverlays(
+    fromLandmarks landmarks: [[NormalizedLandmark]],
```
Comment (priankakariatyml) [inline examples/face_landmarker/ios/FaceLandmarker/Views/FaceOverlay.swift:212]: s/landmarks/multipleFaceLandmarks

File: examples/face_landmarker/ios/FaceLandmarker/Views/FaceOverlay.swift [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+      guard !landmarks.isEmpty else {
+        return []
+      }
+
+      let offsetsAndScaleFactor = OverlayView.offsetsAndScaleFactor(
+        forImageOfSize: originalImageSize,
+        tobeDrawnInViewOfSize: ovelayViewSize,
+        withContentMode: imageContentMode)
+
+      for landmark in landmarks {
```
Comment (priankakariatyml) [inline examples/face_landmarker/ios/FaceLandmarker/Views/FaceOverlay.swift:229]: s/landmark/faceLandmarks

File: examples/face_landmarker/ios/FaceLandmarker/Views/FaceOverlay.swift [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+        var transformedLandmark: [CGPoint]!
```
Comment (priankakariatyml) [inline examples/face_landmarker/ios/FaceLandmarker/Views/FaceOverlay.swift:230]: s/transformedLandmark/transformedFaceLandmarks

File: examples/face_landmarker/ios/FaceLandmarker/Services/FaceLandmarkerService.swift [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+  private func imageGenerator(with videoAsset: AVAsset) -> AVAssetImageGenerator {
+    let generator = AVAssetImageGenerator(asset: videoAsset)
+    generator.requestedTimeToleranceBefore = CMTimeMake(value: 1, timescale: 25)
+    generator.requestedTimeToleranceAfter = CMTimeMake(value: 1, timescale: 25)
+    generator.appliesPreferredTrackTransform = true
+
+    return generator
+  }
+
+  private func detectObjectsInFramesGenerated(
```
Comment (priankakariatyml) [inline examples/face_landmarker/ios/FaceLandmarker/Services/FaceLandmarkerService.swift:212]: s/detectObjectsInFramesGenerated/detectFaceLandmarksInFramesGenerated

**Review by priankakariatyml — APPROVED** [review]

**Pull request merged**